static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -291,30 +300,74 @@ function WidgetBuilder({
 
     if (isEditing && isValidWidgetIndex) {
       const widgetFromDashboard = filteredDashboardWidgets[widgetIndexNum];
+      const newDisplayType =
+        widgetBuilderNewDesign && widgetFromDashboard.displayType === DisplayType.TOP_N
+          ? DisplayType.AREA
+          : widgetFromDashboard.displayType;
+      let queries = normalizeQueries({
+        displayType: newDisplayType,
+        queries: widgetFromDashboard.queries,
+        widgetType: widgetFromDashboard.widgetType ?? WidgetType.DISCOVER,
+        widgetBuilderNewDesign,
+      });
+      let newLimit = widgetFromDashboard.limit;
+
+      if (
+        widgetBuilderNewDesign &&
+        widgetFromDashboard.displayType === DisplayType.TOP_N
+      ) {
+        newLimit = DEFAULT_RESULTS_LIMIT;
+
+        // Use the last aggregate because that's where the y-axis is stored
+        queries = queries.map(query => ({
+          ...query,
+          aggregates: query.aggregates.length
+            ? [query.aggregates[query.aggregates.length - 1]]
+            : [],
+        }));
+      }
```

Comment:
should we update this logic to only check for the `Top N` 1x?

Reply:
Sure, I wasn't sure if I should do this or split them once because I wanted to make sure the `normalizeQueries` call didn't get the wrong display type and I was trying to only call it once.

I changed the code to instantiate some variables and I fill them in depending on whether the display type is `TOP_N`. I had to duplicate the `normalizeQueries` call though